include/simdjson/concepts.h

```diff
@@ -0,0 +1,23 @@
+#ifndef SIMDJSON_CONCEPTS_H
+#define SIMDJSON_CONCEPTS_H
+#ifdef __has_include
+#if __has_include(<version>)
+#include <version>
+#endif
+#endif
+
+#ifdef __cpp_concepts
+#include <utility>
+#include <vector>
+
+namespace simdjson {
+namespace concepts {
```

Comment:
Nice idea to have a dedicated file to define meaningful concepts! I can see this being heavily used later on (UserDefinedType, STL containers, etc...)